src/hooks/useEntry.ts

```diff
@@ -29,18 +29,49 @@ export const useEntry = (journalId: string) => {
   if (!journalId) {
     throw new Error('journalId is required to use useEntry.');
   }
+
+  const flashContext = useContext(FlashContext);
+  
+  if (!flashContext) {
+    throw new Error('useEntry must be used within a FlashProvider');
+  }
+
+  const { processFlash } = flashContext;
+
+  /**
+   * Processes API errors and extracts flash messages
+   * @param error - The API error
+   * @returns The extracted flash message or a default error message
+   */
+  const handleError = (error: any) => {
+    try {
+      const errorResponse = JSON.parse(error.message);
+      return errorResponse?.flash ? errorResponse : { flash: { error: ['An unexpected error occurred.'] } };
+    } catch {
+      return { flash: { error: ['An unexpected error occurred.'] } };
+    }
+  };
   
   /**
    * @typeParam T - The type of the response data
    * @param requestBundle - Bundle containing endpoint and request options
    * @returns A Promise resolving to the response data or an error
    */
   const request = async <T>(requestBundle: RequestBundle): Promise<T> => {
-    const entryUri = `${ENTRY_BASE_URL}/${journalId}/entries${requestBundle.endpoint}`;
+    const entryUri = `${process.env.NEXT_PUBLIC_ENTRY_BASE_URL}/${journalId}/entries${requestBundle.endpoint}`;
+
```

Comment:
ENTRY_BASE_URL needs to be checked similar to suggestion made in `useAccess.ts`.

```suggestion
    const entryUri = `${ENTRY_BASE_URL}/${journalId}/entries${requestBundle.endpoint}`;
```